### PR TITLE
E2E tests

### PR DESCRIPTION
**Pull request by niklabh**

### CODE REVIEW

**Comment by niklabh** [comment]
Please review @Maar-io

File: contracts/lottery/Cargo.toml [hunk 1]
```diff
@@ -4,6 +4,9 @@ version = "0.1.0"
 authors = ["Nikhil Ranjan"]
 edition = "2021"
 
+[dev-dependencies]
+ink_e2e = "4.0.1"
```
Comment (Maar-io) [inline contracts/lottery/Cargo.toml:8]: you can try with 4.2

File: contracts/lottery/Cargo.toml [hunk 2]
```diff
@@ -4,6 +4,9 @@ version = "0.1.0"
...
+
 [dependencies]
 ink = { version = "~4.1.0", default-features = false }
```
Comment (Maar-io) [inline contracts/lottery/Cargo.toml:11]: 4.2?

File: contracts/lottery/src/lib.rs [hunk 3]
```diff
@@ -118,17 +118,21 @@ mod lottery {
         #[ink(message, payable)]
         pub fn enter(&mut self) -> Result<()> {
             if !self.running {
+                ink::env::debug_println!("here1");
```
Comment (Maar-io) [inline contracts/lottery/src/lib.rs:121]: do you need these 3 println?

File: contracts/lottery/src/lib.rs [hunk 4]
```diff
@@ -203,4 +207,142 @@ mod lottery {
             assert_eq!(lottery.random() > 0, true);
         }
     }
+
+    /// end-to-end (E2E) or integration tests for lottery.
+    ///
+    /// When running these you need to make sure that you:
+    /// - Compile the tests with the `e2e-tests` feature flag enabled (`--features e2e-tests`)
+    /// - Are running a Substrate node which contains `pallet-contracts` in the background
```
Comment (Maar-io) [inline contracts/lottery/src/lib.rs:215]: this is not true. you need to have the ENV variable set

**Review by Maar-io — APPROVED** [review]
LGTM

**Pull request merged**